[docs] archive deprecation (#21994)
MystenLabs/sui: docs/content/guides/operator/archives.mdx

> description: The archive is a historical record of all transactions on Sui. Enable archiving on your Full nodes as a best practice.

Reviewer: Not related to this doc, but would be good to add a simple info to this doc section as well. -https://docs.sui.io/guides/operator/data-management#sui-archival-data, to remove everything after the first paragraph there, and just direct / link folks to this doc.

---

> A Sui archive is a history of all transaction data on Sui. As a Sui node operator, you can write this history to an object store like S3, GCS, or similar for safe keeping. Saving this archive data is considered a best practice because Sui prunes transactions on Full nodes to remove historical transactions and their effects. Peer nodes, for example, might not catch up with all transactions and effects through synchronization if they lag behind the current epoch by more than the latest few epochs. Instead of relying on synchronization, peer nodes can fallback to downloading this information from an archive.

Reviewer: ```suggestion
A Sui archive is a history of all transaction data on Sui. In some cases, peer nodes may not catch up with all transactions and effects through synchronization if they lag behind the current epoch by more than the latest few epochs. In such cases, instead of relying on synchronization, peer nodes can fallback to downloading the relevant information from an archive.
```

Reviewer: Basically removing the part related to writing the archive

A Sui archive is a history of all transaction data on Sui. As a Sui node operator, you can write this history to an object store like S3, GCS, or similar for safe keeping. Saving this archive data is considered a best practice because Sui prunes transactions on Full nodes to remove historical transactions and their effects. Peer nodes, for example, might not catch up with all transactions and effects through synchronization if they lag behind the current epoch by more than the latest few epochs. Instead of relying on synchronization, peer nodes can fallback to downloading this information from an archive.

> Sui Archival nodes (Full nodes that write to an archive) don't store historical state on local storage and don't help query historical data. They serve the purpose of enabling peer nodes to catch up to the latest checkpoint and are useful for auditing and verifying the complete history of all transactions on the network.

Reviewer: I think we shouldn't remove this right away, but instead should move it to a new section towards the end, which could be named `Legacy archival process (deprecated)` or something similar. And then after a couple of releases, we could totally remove it.

> ## Create an Archival node

Reviewer: Same as above (for this entire "Create an Archival node" section):

I think we shouldn't remove this right away, but instead should move it to a new section towards the end, which could be named `Legacy archival process (deprecated)` or something similar. And then after a couple of releases, we could totally remove it.

Author: The 'Create an Archival node' part doesn't work and should have been removed a long time ago

Reviewer: Got it, thanks.

> Starting with the 1.49 release, the existing special archive format is being deprecated in favor of a generic data ingestion format. Please migrate your configs using the configuration example listed below

Reviewer: ```suggestion
Starting with the 1.49 release, the existing special archive format is being deprecated in favor of a generic data ingestion format. Please migrate your configs using the configuration example listed below. The deprecation configuration is mentioned below for reference, but will be removed after a few releases.
```

> ## Set up archival fallback {#set-up-archival-fallback}

Reviewer: Again, I think we should move the current documentation for fallback to a section like `Legacy archival process (deprecated)` or something similar. And then after a couple of releases, we could totally remove it.

The new section referring to the checkpoint buckets could be named `Set up archival fallback (latest)` or similar, and then once we remove the legacy portion after a couple of releases, we could remove the `(latest)` from here.

description: The archive is a historical record of all transactions on Sui. Enable archiving on your Full nodes as a best practice.
---

A Sui archive is a history of all transaction data on Sui. In some cases, peer nodes may not catch up with all transactions and effects through synchronization if they lag behind the current epoch by more than the latest few epochs. In such cases, instead of relying on synchronization, peer nodes can fallback to downloading the relevant information from an archive.

:::info

> Starting with the 1.49 release, the existing special archive format is being deprecated in favor of a generic data ingestion format. Please migrate your configs using the configuration example listed below. The deprecation configuration is mentioned below for reference, but will be removed after a few releases.

Reviewer 2: ```suggestion
Starting with the 1.49 release, the existing special archive format is being deprecated in favor of a generic data ingestion format. Please migrate your configs using the configuration example that follows. The deprecated configuration is currently included for reference, but will be removed after a few releases.
```

:::

> ## Set up archival fallback(latest) {#set-up-archival-fallback}

Reviewer 2: ```suggestion
## Set up archival fallback (latest) {#set-up-archival-fallback}
```

To enable your node to fallback to an archive in case of lag, add this to your fullnode.yaml file:

<Tabs groupId="object-store">
<TabItem value="mainnet" label="Mainnet">

```yaml
state-archive-read-config:
  - ingestion-url: "https://checkpoints.mainnet.sui.io"
    # How many objects to read ahead when catching up
    concurrency: 5
```

</TabItem>
<TabItem value="testnet" label="Testnet">

```yaml
state-archive-read-config:
  - ingestion-url: "https://checkpoints.testnet.sui.io"
    # How many objects to read ahead when catching up
    concurrency: 5
```

</TabItem>
</Tabs>

> ## Set up archival fallback(deprecated) {#set-up-archival-fallback}

Reviewer: ```suggestion
## Set up archival fallback(deprecated) {#set-up-archival-fallback-deprecated}
```
